orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py

```diff
@@ -204,68 +232,27 @@ def _export_pt_1_10(g, n, *args, **kwargs):
 def _post_process_after_export(exported_model, enable_custom_autograd_function):
     if enable_custom_autograd_function:
         return _post_process_enabling_autograd_fallback(exported_model)
-
-    is_pythonop_needed = False
-    for node in exported_model.graph.node:
-        if node.domain == "com.microsoft" and node.op_type in ["PythonOp"]:
-            is_pythonop_needed = True
-            break
-
-    if is_pythonop_needed:
-        warnings.warn(
-            "Detected autograd functions usage in current model, the run will fail \
-                      without enabling '_enable_custom_autograd_function'. Please enable it with: \
-                      'module._execution_manager(is_training_mode)._enable_custom_autograd_function = True'",
-            UserWarning,
-        )
-
     return exported_model
 
 
 def _post_process_enabling_autograd_fallback(exported_model):
-    registered_name_mappings = {}
-    skipped_autograd_function_list = ortmodule._defined_from_envvar("ORTMODULE_SKIPPED_AUTOGRAD_FUNCTIONS", "").split(
-        ","
-    )
-    for kclass in torch.autograd.Function.__subclasses__():
-        full_qualified_name = _full_name(kclass)
-        if full_qualified_name in skipped_autograd_function_list:
-            continue
-        # Collect mapping of class names to full qualified class names.
-        if kclass.__name__ not in registered_name_mappings:
-            registered_name_mappings[kclass.__name__] = []
-        registered_name_mappings[kclass.__name__].append(full_qualified_name)
-
-        # Register function with class names.
-        register_torch_autograd_function(kclass.__name__, kclass)
```

Comment:
seems unrelated to the PR goal. Is that intentional?

Reply:
Oh, this is by intention, I mentioned it a bit in the last paragram "With the above change, we don't need use kwargs[name] and kwargs[module] , and don't need check naming conflicting or ORTMODULE_SKIPPED_AUTOGRAD_FUNCTIONS env var any more.". 

Since we can get the autograd.Function classes in the exporter function, are using full qual name now, we can just do the registration there, don't need do this after exporting. There is a side good effect: for those not used torch.autograd.FUnction.__subclass__(), we don't register them any more.

Reply:
BTW, register_torch_autograd_function tolerant repeated registry, it respect the first time registration behavior. ;)